crates/router/src/connector/iatapay/transformers.rs

```diff
@@ -225,7 +229,7 @@ impl<F, T>
                     mandate_reference: None,
                     connector_metadata: None,
                     network_txn_id: None,
-                    connector_response_reference_id: None,
+                    connector_response_reference_id: connector_response_reference_id.clone(),
```

Comment:
```suggestion
                    connector_response_reference_id: item.response.merchant_payment_id.or(item.response.iata_payment_id),
```
Here, `connector_response_reference_id` should have `merchant_payment_id`, if `merchant_payment_id` is not present, then on fallback we should send `iata_payment_id`

Reply:
Sorry I had misunderstood the task. I have set iata_payment_id as the primary response and merchant_payment_id as the fallback. Shall i do it the other way around in the connector_response_reference_id logic ( as connector_response_reference_id is used again in line no. 245) or directly commit the suggested change?